packages/server/src/main/java/software/uncharted/terarium/hmiserver/proxies/knowledge/KnowledgeMiddlewareProxy.java

```diff
@@ -101,16 +101,18 @@ ResponseEntity<JsonNode> postProfileDataset(
 	/**
 	 * Transform source code to AMR
 	 *
-	 * @param codeId      (String): id of the code artifact
-	 * @param name        (String): the name to set on the newly created model
-	 * @param description (String): the description to set on the newly created model
+	 * @param codeId       (String): id of the code artifact
+	 * @param name         (String): the name to set on the newly created model
+	 * @param description  (String): the description to set on the newly created model
+	 * @param dynamicsOnly (Boolean): whether to only run the amr extraction over specified dynamics from the code object in TDS
 	 * @return (ExtractionResponse)
 	 */
 	@PostMapping("/code_to_amr")
 	ResponseEntity<ExtractionResponse> postCodeToAMR(
 		@RequestParam("code_id") String codeId,
 		@RequestParam("name") String name,
-		@RequestParam("description") String description
+		@RequestParam("description") String description,
```

Comment:
I think there may be an issue here with the controller having these marked as optional but the proxy having them marked as required.  The Controller should also define a default value (part of the @RequestParam), or the proxy should also be updated to mark `name`, `description` and `dynamicsOnly` as not required